src/main/java/net/floodlightcontroller/websocket/WebSocketManager.java

```diff
@@ -0,0 +1,257 @@
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.websocket.api.Session;
+
+import net.floodlightcontroller.core.module.FloodlightModuleContext;
+import net.floodlightcontroller.core.module.FloodlightModuleException;
+import net.floodlightcontroller.core.module.IFloodlightModule;
+import net.floodlightcontroller.core.module.IFloodlightService;
+
+import net.floodlightcontroller.restserver.IRestApiService;
+import net.floodlightcontroller.storage.IResultSet;
+import net.floodlightcontroller.storage.IStorageSourceListener;
+import net.floodlightcontroller.storage.IStorageSourceService;
+import net.floodlightcontroller.storage.StorageException;
+
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+
+
+/**
+ * Module starts a Jetty server, and registers storage listeners for real-time table changes
+ *
+ * TODO: Check to make sure all threads are finished correctly.
+ *
+ * @author luke.sanderson@carleton.ca
+ */
+public class WebSocketManager implements IStorageSourceListener, IFloodlightModule {
+	
+	// Class logger
+	private static final Logger logger = LoggerFactory.getLogger(WebSocketManager.class);
+	private static final boolean VERBOSE = false;
+	
+	private static WebSocketManager instance;
+	protected IStorageSourceService storageSourceService;
+	protected IRestApiService restApiService;
+	private HashMap<Session, List<String>> activeTables= new HashMap<Session, List<String>>();
+	private List<Session> sessions = new ArrayList<Session>();	
+	
+    /** Method to keep instance static **/
+    public static WebSocketManager getInstance() {
+        return instance;
+    }
+
+	@Override
+	public Collection<Class<? extends IFloodlightService>> getModuleServices() {
+		
+		return null;
+	}
+
+	@Override
+	public Map<Class<? extends IFloodlightService>, IFloodlightService> getServiceImpls() {
+		
+		return null;
+	}
+
+	@Override
+	public Collection<Class<? extends IFloodlightService>> getModuleDependencies() {
+		
+		
+		Collection<Class<? extends IFloodlightService>> l =
+				new ArrayList<Class<? extends IFloodlightService>>();
+	
+		l.add(IStorageSourceService.class);
+		l.add(IRestApiService.class);
+		return l;
+	}
+
+	@Override
+	public void init(FloodlightModuleContext context) throws FloodlightModuleException {
+		storageSourceService = context.getServiceImpl(IStorageSourceService.class);
+		restApiService = context.getServiceImpl(IRestApiService.class);
+	    if(VERBOSE){logger.info("made it to init");}
+	}
+
+	@Override
+	public void startUp(FloodlightModuleContext context) throws FloodlightModuleException {
+		
+		instance = this; //singleton to assure one is being called
+		if(VERBOSE){logger.info("made it before startup");}
+		
+		Server server = new Server();
+		ServerConnector connector = new ServerConnector(server);
+		connector.setPort(8111);
```

Comment:
Can we make this configurable, like there's a properties file that some modules use, for example the stats module can be enabled by default.

If we can add that there that would be helpful.